Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -164,6 +176,52 @@ public struct NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Sendable {
             outboundWriter: outboundWriter
         )
     }
+
+    /// Provides scoped access to the inbound and outbound side of the underlying ``Channel``.
+    ///
+    /// - Important: After this method returned the underlying ``Channel`` will be closed.
+    ///
+    /// - Parameter body: A closure that gets scoped access to the inbound and outbound.
+    public func withInboundOutbound<Result>(
```

Comment:
Hm why? The only type called stream is the `InboundStream`. I was going between `withInboundStreamAndOutboundWriter` before but decided to follow the same logic where we renamed them to `inbound` and `outbound`

Reply:
I just don't think the name is particularly clear. 

I'm not sure `withStreams` is the right name either. I suggested "streams" because it's how I'd refer to `inbound` and `outbound` collectively.

Maybe the naming should indicate the semantics of what the function offers rather than what the closure provides to the caller? Thinking along the lines of "with auto closing streams", "with lifecycle managed streams" (I don't like these _either_ but throwing them out anyway...)